File: ViewModelsLayer/ViewModels/GameViewModels/RequestGameViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;
using ViewModelsLayer.ViewModels.UserViewModels;

namespace ViewModelsLayer.ViewModels.GameViewModels
{
    public class RequestGameViewModel
    {
        [Required]
        public RequestUserViewModel User { get; set; }
        [Required]
        [Range(0, 4, ErrorMessage = "Invalid bots quantity")]
        public int BotQuantity { get; set; }
        [Required]
        [Range(0, 15, ErrorMessage = "Invalid rounds quantity")]
        public int RoundQuantity { get; set; }
        [Required]
        [Range(1, 1000000, ErrorMessage = "Invalid rate")]
        public int UserRate { get; set; }
    }
}

File: API/Angular/src/app/viewmodels/GameViewModels/request-game-view-model.ts
import { RequestUserViewModel } from '../UserViewModels/request-user-view-model';

export default class RequestGameViewModel
{
  constructor(user, botQuantity, roundQuantity, userRate)
  {
    this.user = user;
    this.botQuantity = botQuantity;
    this.roundQuantity = roundQuantity;
    this.userRate = userRate;
  }

  public user?: RequestUserViewModel;
  public botQuantity: number;
  public roundQuantity: number;
  public userRate: number;
}
